content/v2.0/reference/storage-engine.md

```diff
@@ -0,0 +1,157 @@
+Batches of [points](/v2.0/reference/glossary/#point) are sent to InfluxDB, compressed, and written to a WAL for immediate durability.
+The points are also written to an in-memory cache and become immediately queryable.
+The cache is periodically written to disk in the form of [TSM](#time-structured-merge-tree-tsm) files.
+As TSM files accumulate, they are combined and compacted into higher level TSM files.
+
+Points can be sent individually; however, for efficiency, most applications send points in batches.
+A typical batch ranges in size from hundreds to thousands of points.
+Points in a POST body can be from an arbitrary number of series, measurements, and tag sets.
+Points in a batch do not have to be from the same measurement or tagset.
+
+## Write Ahead Log (WAL)
+
+The Write Ahead Log (WAL) ensures durability by retaining data when the storage engine restarts.
+It ensures that written data does not disappear in an unexpected failure.
+When a client sends a write request, the following occurs:
+
+1. Write request is appended to the end of the WAL file.
+2. `fsync()` the data to the file.
+3. Update the in-memory database.
+   <!-- 3. Update the in-memory CACHE? -->
+4. Return success to caller.
+
+`fsync()` takes the file and pushes pending writes all the way through any buffers and caches to disk.
+As a system call, `fsync()` has a kernel context switch which is expensive _in terms of time_ but guarantees your data is safe on disk.
+
+{{% note%}}
+To `fsync()` less frequently, batch your points (send in ~2000 points at a time).
+{{% /note %}}
+
+When the storage engine restarts, the WAL file is read back into the in-memory database.
+InfluxDB then snswer requests to the `/read` endpoint.
+
+<!-- ===== V1 material -->
+<!-- TODO is this still true? -->
+<!-- On the file system, the WAL is made up of sequentially numbered files (`_000001.wal`). -->
+<!-- The file numbers are monotonically increasing and referred to as WAL segments. -->
+<!-- When a segment reaches 10MB in size, it is closed and a new one is opened. Each WAL segment stores multiple compressed blocks of writes and deletes. -->
+<!-- Each entry in the WAL follows a [TLV standard](https://en.wikipedia.org/wiki/Type-length-value) with a single byte representing the type of entry (write or delete), a 4 byte `uint32` for the length of the compressed block, and then the compressed block. -->
+
+{{% note%}}
+Once you receive a response to a write request, your data is on disk!
+{{% /note %}}
+
+## Cache
+
+Queries to the storage engine will merge data from the Cache with data from the TSM files.
+Queries execute on a copy of the data that is made from the cache at query processing time.
+This way writes that come in while a query is running won’t affect the result.
+
+The cache is an in-memory copy of data points current stored in the WAL.
+Points are organized by the key, which is the measurement, tag set, and unique field.
```

Comment:
Points are organize by series. A series key defines the contents of a series and is comprised of a measurement, tag set, and field key.